lib/client/index.js

```diff
@@ -101,16 +101,14 @@ function manage(datasetId, options, queryParams, metaData) {
     return deferred.promise;
   }
 
-  //If there is already a manager initialising for this dataset, return the promise.
-  if (dataSetManagerPromises[datasetId]) {
-    return dataSetManagerPromises[datasetId];
-  }
-
-  var deferredManagerDeferred = q.defer();
-  dataSetManagerPromises[datasetId] = deferredManagerDeferred.promise;
-
   //Using the $fh.sync API to manage a single data set with the passed parameters above.
   $fh.sync.manage(datasetId, options, queryParams, metaData, function() {
+
+    //If the manager for this data set already exists, no need to create a new one.
+    if (dataSetManagers[datasetId]) {
+      return deferred.resolve(dataSetManagers[datasetId]);
```

Comment:
Ah, I see now that it's only short-circuiting. This is still refactorable to A+, so :+1: